pkg/controller/reconciler/reconciler.go

```diff
@@ -253,6 +253,12 @@ func (r *Reconciler) apply(ctx context.Context, res common.PlatformObject) error
 		Manifests:  make([]types.ManifestInfo, 0),
 	}
 
+	// reset conditions so any unknown condition eventually set on
+	// the owned resource get cleaned up. This is the case when a
+	// condition is replaced/removed.
+
+	rr.Conditions.Reset()
```

Comment:
this is going to be called everytime by Reconcile()

Reply:
yes,

Reply:
once we are in 2026.5 we can safely remove this part, as no downstream upgrade can be performed from pre-2.17